### ISSUE TITLE
Multi-thread bug in rake task

### ISSUE THREAD

**Issue by MaximeRDY** [rails/rails#22097]
Hello guys, 

I use `Rails 3.2.22` and  the gem [Parallel](https://github.com/grosser/parallel). And I think I'm gonna be crazy because of class loading. 

I use [DelayedJob](https://github.com/tobi/delayed_job) background job on **Heroku** and randomly some of my class is not defined and I have this kind of error `NameError: uninitialized constant  RequestAvailabilityForRoutes`. So the first thing I've checked is to know if my class was well defined. So I've tried with no multi-thread, it seems working, but I want to use multi-thread.. 

Next, I've tried to change my autoload paths and follow this [tutorial] (http://hakunin.com/rails3-load-paths)... Of course, no effect appeared... 

And finally I find this issue https://github.com/rails/rails/issues/13142. But I don't understand how to not fallback in non-thread-safe autoloading .. 

My configuration in **production** is : 
```
  config.threadsafe!
  config.dependency_loading = true if $rails_rake_task
``` 

What I'm doing wrong ? Is the gem I use ? There is a workaround I didn't know ? 

Please help me, I'm a little desperate..

**Comment by matthewd** [comment]
Please use the mailing list or StackOverflow for questions/help, where a wider community will be able to help you. We reserve the issues tracker for issues only.

**Comment by matthewd** [comment]
(Also, Rails 3.2 is no longer supported.)

**Comment by zdennis** [comment]
@MaximeRDY, I just posted an update to https://github.com/rails/rails/issues/13142 . The erroneous commit got reverted, but it didn't get back ported so it looks to only be available in v5.0.0.beta1 and later.
